### DISCUSSION TITLE
Support optional keyfinder-cli params in keyfinder plugin

### DISCUSSION THREAD

**Discussion by johnhamelink** [beetbox/beets#4097] [Ideas]
keyfinder-cli supports multiple key formats. The default is fine, but for DJs and other people who rely on quickly understanding the "compatibility" between tracks without a background in traditional music theory, other formats may be more useful.

All that's required here is to modify the plugin to add an extra optional configuration item where flags can be passed to the keyfinder-cli process. This would allow customisation of the configuration used by keyfinder-cli when it's run from within beets.

Thanks for your consideration!
JH

**Comment by sampsyo** [comment]
That sounds reasonable to me! Maybe it would be useful to jot down what the common options for different formats would look like?

**Reply by johnhamelink** [reply]
Currently keyfinder-cli only provides one flag:

```
-n [standard|openkey|camelot]
```

**Reply by sampsyo** [reply]
Got it. So maybe we'd even want to provide a somewhat higher-level option, `format`, that just passes its argument through to the `-n` flag.

**Reply by johnhamelink** [reply]
Yeah that's probably the nicest option in terms of configuration UX!

**Comment by johnhamelink** [comment]
I ended up coming back to this!

I've written a little bash script which provides the correct parameters for keyfinder-cli:

Saved in `$PATH` as `keyfinder-camelot.sh`:

```bash
#!/usr/bin/bash

file=("${@:2}")
# I'm not sure why, but the * is the magic sauce to make this work.
keyfinder-cli -n camelot "${file[*]}"
```

Then in `config.yaml`:

```yaml
keyfinder:
  bin: keyfinder-camelot.sh
```
